src/extension.ts

```diff
@@ -134,6 +134,15 @@ export function activate(context: vscode.ExtensionContext) {
 	registerCommand('cosmosDB.api.getDatabase', async () => {
 		return (<IAzureParentNode>await tree.showNodePicker([MongoDatabaseTreeItem.contextValue, DocDBDatabaseTreeItem.contextValue])).id;
 	});
+	registerCommand('cosmosDB.api.getConnectionString', async (treeItemId: string) => {
+		const node = (<IAzureNode>await tree.findNode(treeItemId));
```

Comment:
Don't cast this to `IAzureNode`. The type of `IAzureNode | undefined` is correct. You need to handle the case where it's undefined (aka it didn't find the node). Maybe just throw an error

Reply:
I'll add that throwing an error would show up to the user as stemming from the Cosmos extension instead of the caller function. The report error button would also submit an error to cosmos - so we'll have to look here to see how the feature is doing.

Reply:
Good observation.  That's more reason to use an API format rather than commands.